File: DSA/Array/Hourglass.cpp
#include<iostream>
using namespace std;
int main()
{
    int n,m,sum=0,max=0;
    cin>>n>>m;
    cout<<n<<" "<<m;
    int arr[n][m];
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        cin>>arr[i][j];
    }
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        cout<<arr[i][j]<<" ";
    }
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; i < n; j++)
        {
            for (int k = i; k <i+3 && k<n ; k++)
            {
                for (int l = j; l < j+3 && l<m; l++)
                {
                    sum+=arr[k][l];
                }
                
            }
            if(max<sum)
            {
                max=sum;
            }
        }
        
    }
    cout<<"max= "<<max;
    return 0;
    
}

File: DSA/Array/Practice Python/HourGlass.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the hourglassSum function below.
def hourglassSum(arr):
    li=[]
    for i in range(4):
        for j in range(4):
            count=0
            add=0
            for k in range(j,j+3):
                add+=arr[i][k]+arr[i+2][k]
                count+=1
                if count==2:
                    add+=arr[i+1][k]

            li.append(add)
    return max(li)
            

            

if __name__ == '__main__':
    
    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    result = hourglassSum(arr)

    print(result)